sdk/identity/identity/src/credentials/defaultAzureCredentialFunctions.ts

```diff
@@ -0,0 +1,165 @@
+  const workloadFile = process.env.AZURE_FEDERATED_TOKEN_FILE;
+  const tenantId = options?.tenantId ?? process.env.AZURE_TENANT_ID;
+  if (managedResourceId) {
+    const managedIdentityResourceIdOptions: ManagedIdentityCredentialResourceIdOptions = {
+      ...options,
+      resourceId: managedResourceId,
+    };
+    return new ManagedIdentityCredential(managedIdentityResourceIdOptions);
+  }
+
+  if (workloadFile && workloadIdentityClientId) {
+    const workloadIdentityCredentialOptions: DefaultAzureCredentialOptions = {
+      ...options,
+      tenantId: tenantId,
+    };
+
+    return new ManagedIdentityCredential(
+      workloadIdentityClientId,
+      workloadIdentityCredentialOptions,
+    );
+  }
+
+  if (managedIdentityClientId) {
+    const managedIdentityClientOptions: ManagedIdentityCredentialClientIdOptions = {
+      ...options,
+      clientId: managedIdentityClientId,
+    };
+
+    return new ManagedIdentityCredential(managedIdentityClientOptions);
+  }
+
+  // We may be able to return a UnavailableCredential here, but that may be a breaking change
+  return new ManagedIdentityCredential(options);
+}
+
+/**
+ * Creates a {@link WorkloadIdentityCredential} from the provided options.
+ * @param options - Options to configure the credential.
+ *
+ * @internal
+ */
+export function createDefaultWorkloadIdentityCredential(
+  options?: DefaultAzureCredentialOptions | DefaultAzureCredentialClientIdOptions,
+): TokenCredential {
+  const managedIdentityClientId =
+    (options as DefaultAzureCredentialClientIdOptions)?.managedIdentityClientId ??
+    process.env.AZURE_CLIENT_ID;
+  const workloadIdentityClientId =
+    (options as DefaultAzureCredentialClientIdOptions)?.workloadIdentityClientId ??
+    managedIdentityClientId;
+  const workloadFile = process.env.AZURE_FEDERATED_TOKEN_FILE;
+  const tenantId = options?.tenantId ?? process.env.AZURE_TENANT_ID;
+  if (workloadFile && workloadIdentityClientId) {
+    const workloadIdentityCredentialOptions: WorkloadIdentityCredentialOptions = {
+      ...options,
+      tenantId,
+      clientId: workloadIdentityClientId,
+      tokenFilePath: workloadFile,
+    };
+    return new WorkloadIdentityCredential(workloadIdentityCredentialOptions);
+  }
+  if (tenantId) {
+    const workloadIdentityClientTenantOptions: WorkloadIdentityCredentialOptions = {
+      ...options,
+      tenantId,
+    };
+    return new WorkloadIdentityCredential(workloadIdentityClientTenantOptions);
+  }
+
+  // We may be able to return a UnavailableCredential here, but that may be a breaking change
+  return new WorkloadIdentityCredential(options);
+}
+
+/**
+ * Creates a {@link AzureDeveloperCliCredential} from the provided options.
+ * @param options - Options to configure the credential.
+ *
+ * @internal
+ */
+export function createDefaultAzureDeveloperCliCredential(
+  options: DefaultAzureCredentialOptions = {},
+): TokenCredential {
+  return new AzureDeveloperCliCredential({ ...options });
```

Comment:
This is redundant, please fix all other places that do this
```suggestion
  return new AzureDeveloperCliCredential(options);
```